bundle/edu.gemini.spModel.io/src/main/scala/edu/gemini/spModel/io/impl/migration/to2015A/To2015A.scala

```diff
@@ -91,8 +81,8 @@ object To2015A {
 
         // Add them back as children of this template parameters
         // data object ParamSet.
-        tp.addParamSet(targetMap(targetId).getParamSet(fact))
-        tp.addParamSet(condsMap(condsId).getParamSet(fact))
+        tp.addParamSet(PioXmlUtil.read(targetMap(targetId)).asInstanceOf[ParamSet])
```

Comment:
Unfortunately `PioXmlUtil.read` could return any `PioNode` subclass.  I only know it is safe here because in the `targetMap` I've stuffed `ParamSet` strings in the `mapParamSets` function above.  This is all suboptimal.